SmartDeviceLinkTests/SDLMenuUpdateAlgorithmSpec.m

```diff
@@ -0,0 +1,147 @@
+//
+//  SDLMenuUpdateAlgorithmSpec.m
+//  SmartDeviceLinkTests
+//
+//  Created by Justin Gluck on 5/15/19.
+//  Copyright © 2019 smartdevicelink. All rights reserved.
+//
+
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+#import <OCMock/OCMock.h>
+
+#import "SDLMenuCell.h"
+#import "SDLMenuRunScore.h"
+#import "SDLMenuUpdateAlgorithm.h"
+
+QuickSpecBegin(SDLMenuUpdateAlgorithmSpec)
```

Comment:
Not tested in here is:

1. When there are old menu cells but no new cells
2. No old cells and no new cells (is this case even possible?)
3. When the run loop index becomes greater than the number of `oldMenuCells` (is this possible?)

Reply:
3. this would cause the loop to stop would it not?

Reply:
added case 1. I don't think 2 is possible unless the developer sets menu = @[] 2 times in a row